pkg/ebpfsyncer/ebpfsyncer_test.go

```diff
@@ -0,0 +1,558 @@
+		t.Log(err)
+	}
+}
+
+func beforeEach(t *testing.T) {
+	// First, check if the user is root; skip otherwise.
+	fmt.Println("Checking if this test runs with sufficiently high privileges")
+	currentUser, err := user.Current()
+	if err != nil {
+		t.Fatalf("Unable to get current user: %s", err)
+	}
+	if currentUser.Uid != "0" {
+		t.Skipf("Skipping this test due to insufficient privileges")
+	}
+
+	// Generate test interfaces and populate the slice. Then, run the cleanup (that way, we can delete the interfaces
+	// if they already exist).
+	fmt.Println("By populating the interface list")
+	interfaces = []netlink.Veth{}
+	for i := 0; i < numTestInterfaces; i++ {
+		// Create interface.
+		la := netlink.NewLinkAttrs()
+		la.Name = fmt.Sprintf("%s%d", interfacePrefix, i)
+		tap := netlink.Veth{
+			LinkAttrs: la,
+			PeerName:  fmt.Sprintf("%s-peer", la.Name),
+		}
+		interfaces = append(interfaces, tap)
+	}
+
+	// Run a cleanup, just in case (cleanup will only log failures to clean up, it will not fail).
+	fmt.Println("By running a cleanup")
+	cleanup(t)
+
+	fmt.Println("By creating the namespace")
+	cmd := exec.Command("ip", "netns", "add", netnsName)
+	if err := cmd.Run(); err != nil {
+		t.Fatal(err)
+	}
+	cmd = exec.Command("ip", "netns", "exec", netnsName, "ip", "link", "set", "dev", "lo", "up")
+	if err := cmd.Run(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Now, actually create the interfaces.
+	fmt.Println("By creating new test interfaces for testing")
+	for i, tap := range interfaces {
+		err = netlink.LinkAdd(&tap)
+		if err != nil {
+			t.Fatalf("Could not add link %s, err: %q", tap.Name, err)
+		}
+		err = netlink.LinkSetUp(&tap)
+		if err != nil {
+			t.Fatalf("Could not set link state to up for link %s, err: %q", tap.Name, err)
+		}
+
+		// Assign 192.0.2.x+1/32 to dummyx, and so on.
+		addr0, err := netlink.ParseAddr(fmt.Sprintf("192.0.2.%d/30", i*4+1))
+		if err != nil {
+			t.Fatalf("Could not parse IP address, err: %q", err)
+		}
+		err = netlink.AddrAdd(&tap, addr0)
+		if err != nil {
+			t.Fatalf("Could not add IP address to interface, err: %q", err)
+		}
+
+		fmt.Println("By moving the peer interfaces to their own namespace and by assigning an IP")
+		cmd := exec.Command("ip", "link", "set", "dev", interfaces[i].PeerName, "netns", netnsName)
+		if err := cmd.Run(); err != nil {
+			t.Fatalf("Could not set netns for %s, err: %q", interfaces[i].PeerName, err)
+		}
+		cmd = exec.Command("ip", "netns", "exec", netnsName, "ip", "address", "add", "dev", interfaces[i].PeerName, fmt.Sprintf("192.0.2.%d/30", i*4+2))
+		if err := cmd.Run(); err != nil {
+			t.Fatalf("Could not add IP address to interface %s in netns %s, err: %q", interfaces[i].PeerName, netnsName, err)
+		}
+		cmd = exec.Command("ip", "netns", "exec", netnsName, "ip", "link", "set", "dev", interfaces[i].PeerName, "up")
+		if err := cmd.Run(); err != nil {
+			t.Fatalf("Could not set link to up in netns %s, link %s, err: %q", interfaces[i].PeerName, netnsName, err)
+		}
```

Comment:
is this test in use or dead code ?

Reply:
It's in use. in beforeEach(), this will create a test veth and namespace that's used by each of the Test cases. 
The IP assignments and namespace aren't required for all of them but I think it makes sense to have them in that beforeEach() nevertheless. I intend to improve these specific unit tests over time any way, they are atm the least elaborate part of the tests (lots of copy / paste for example)